components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java

```diff
@@ -109,22 +118,56 @@ public Response apisChangeLifecyclePost(String newState,String publishToGateway,
     }
     @Override
     public Response apisCopyApiPost(String newVersion,String apiId){
-        // do some magic!
-        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.OK, "magic!")).build();
+        boolean isTenantFlowStarted = false;
+        URI newVersionedApiUri = null;
+        APIDTO newVersionedApi = null;
+
+        try {
+            APIIdentifier apiIdentifier = APIMappingUtil.getAPIIdentifier(apiId);
+            String loggedInUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            APIProvider apiProvider = RestApiUtil.getProvider(loggedInUser);
+            String tenantDomain =  CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```

Comment:
The tenants flows have been moved currently from impl level to CXF handler level
Fixed comments in #1091 commit : 897ef52